### PR TITLE
Pass "encoding" and "contype" to "field" event

### PR DESCRIPTION
**Pull request by SamyPesse**
This PR adds the arguments `contype` (Content-Type) and `encoding` to the `field` event, there are already passed to the `file` event.

### CODE REVIEW

**Comment by mscdex** [comment]
I'm curious why this is necessary as the values for these variables will always be the same for non-file parts for `multipart/form-data`.

**Comment by SamyPesse** [comment]
In our case, we are doing a PUT with a file and some metadata, `metadata` is a field with `Content-Type: application/json`:

For example:

```
-----------------------------287032381131322
Content-Disposition: form-data; name="metadata"
Content-Type: application/json

{ "something": "hello world" }
-----------------------------287032381131322
Content-Disposition: form-data; name="datafile2"; filename="g.gif"
Content-Type: image/gif

GIF87a.............,...........D..;
```

We have a middleware that uses `busboy` to parse the request's body. We want it to parse json fields. (or at least be able to know the `content-type`)

**Comment by Marak** [comment]
@SamyPesse - I'm doing this already I think.

Check out these lines:

https://github.com/bigcompany/parse-service-request/blob/master/index.js#L18
https://github.com/bigcompany/parse-service-request/blob/master/index.js#L71

That module is a general streaming request parsing library which uses busboy. It also supports JSON.

**Comment by mscdex** [comment]
@SamyPesse Fair enough. LGTM but can you add readme changes and a test?

**Comment by SamyPesse** [comment]
Sure! :)

**Comment by Marak** [comment]
I think I see the difference now. Sorry for the confusion.

**Comment by SamyPesse** [comment]
@mscdex done! Let me know if it needs changes.

BTW: why not adding the `node_modules` to the `.gitignore` ?

File: README.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ Busboy (special) events
     * Note: if you listen for this event, you should always handle the `stream` no matter if you care about the file contents or not (e.g. you can simply just do `stream.resume();` if you want to discard the contents), otherwise the 'finish' event will never fire on the Busboy instance. However, if you don't care about **any** incoming files, you can simply not listen for the 'file' event at all and any/all files will be automatically and safely discarded (these discarded files do still count towards `files` and `parts` limits).
     * If a configured file size limit was reached, `stream` will both have a boolean property `truncated` (best checked at the end of the stream) and emit a 'limit' event to notify you when this happens.
 
-* **field**(< _string_ >fieldname, < _string_ >value, < _boolean_ >fieldnameTruncated, < _boolean_ >valueTruncated) - Emitted for each new non-file field found.
+* **field**(< _string_ >fieldname, < _string_ >value, < _boolean_ >fieldnameTruncated, < _boolean_ >valueTruncated, < _string_ >encoding, < _string_ >mimetype) - Emitted for each new non-file field found.
```
Comment (mscdex) [inline README.md]: For consistency, it might be better to use the same parameter names as used for the `file` event (`transferEncoding`, and `mimeType`).

**Comment by SamyPesse** [comment]
@mscdex I solved the comment you added.

**Comment by mscdex** [comment]
Landed in d5f4942efe. Thanks!

**Comment by SamyPesse** [comment]
Awesome! Thanks, any plan on releasing a new version on NPM?

**Comment by mscdex** [comment]
Done.

**Pull request closed without merging**